cpp/ql/lib/experimental/semmle/code/cpp/semantic/analysis/SimpleRangeAnalysis.qll

```diff
@@ -108,7 +109,7 @@ predicate convertedExprMightOverflowNegatively(Expr expr) {
  * does not consider the possibility that the expression might overflow
  * due to a conversion.
  */
-predicate exprMightOverflowPositively(Expr expr) { none() }
+predicate exprMightOverflowPositively(Expr expr) { upperBound(expr) > exprMaxVal(expr) }
```

Comment:
I was confused about this for a second: we only generate bounds for expressions that won't overflow (due to the disjunction in the `bounded` predicate), but here it looks like we're looking for expressions that actually do have an overflowing upper bound. However, this works because `upperBound` is defined to only use the constant stage of range-analysis which doesn't (and can't) have that disjunction that rules out overflowing expressions. Is this correct?

Could we write this explicitly in a comment here, or add QLDoc to `upperBound` and `lowerBound` to make this clear?